api/jobs/gears.py

```diff
@@ -119,12 +132,17 @@ def insert_gear(doc):
     if doc.get("invocation-schema"):
         del(doc["invocation-schema"])
 
+    now = datetime.datetime.utcnow()
+
+    doc['created']  = now
+    doc['modified'] = now
+
     config.db.gears.insert(doc)
```

Comment:
This one I feel pretty adamantly about. It doesn't really fix our problem to remember to put bson.ObjectId(<id>) everywhere if we have to put str(<id>) in the opposite situations instead.